pkg/volume/glusterfs/glusterfs.go

```diff
@@ -1045,3 +1056,62 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 	}
 	return &cfg, nil
 }
+
+type glusterfsDiskExpander struct {
+	*glusterfs
+	provisionerConfig
+	spec *v1.PersistentVolume
+}
+
+var _ volume.Expander = &glusterfsDiskExpander{}
+
+func (e *glusterfsDiskExpander) ExpandVolumeDevice(newSize resource.Quantity, oldSize resource.Quantity) error {
+
+	pvSpec := e.spec.Spec
+	glog.V(2).Infof("Request to expand volume: %s ", pvSpec.Glusterfs.Path)
+	volumeName := pvSpec.Glusterfs.Path
+
+	// Fetch the volume for expansion.
+	volumeID := dstrings.TrimPrefix(volumeName, volPrefix)
+
+	//Get details of SC.
```

Comment:
Is fetching storageClass here required? It looks like it is only used for logging. In that case, if you fail to fetch SC, should that maybe be a warning instead of a terminating error?

Reply:
It is also used for extracting heketi url etc from storageClass.

Reply:
its required @saad-ali. We need the REST url from SC of the PV at time of expansion.